components/places/src/storage/bookmarks/public.rs

```diff
@@ -0,0 +1,582 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use super::*;
+use crate::msg_types::BookmarkNode as ProtoBookmark;
+
+/// This type basically exists to become a msg_types::BookmarkNode, but is
+/// slightly less of a pain to deal with in rust.
+#[derive(Debug, Clone)]
+pub struct PublicNode {
```

Comment:
So many wrapper types for bookmarks! 😅 I understand why they all exist, but _gosh_ there's a lot.

Reply:
I'm absolutely planning on filing an issue about it later. I did try to make do with just the protobuf type, but it's a pain. For a lot of them (possibly including this one) I think we need to cut a bit more ruthlessly. Just because  something can be encoded in the type system doesn't mean the benefits that gives us outweigh the costs (cognitive and maintenance) of having to deal with that code